intelmq/bots/experts/splunk_saved_search/expert.py

```diff
@@ -0,0 +1,192 @@
+               combination of:
+               warn: log a warning message
+               send: send the event on unmodified
+               drop: drop the message
+
+    duplicates: list of strings, default [ "warn", "use_first",
+                "send" ], what to do if the search returns more than
+                one result. Any reasonable combination of:
+                warn: log a warning message
+                use_first: use the first search result
+                ignore: do not modify the event
+                send: send the event on
+                drop: drop the message
+
+    overwrite: bool, optional, whether search results replace existing
+               values in the event. If not specified, trying to set an
+               existing field raises intelmq.exceptions.KeyExists.
+
+"""
+
+try:
+    import requests
+except ImportError:
+    requests = None
+
+import intelmq.lib.utils as utils
+from intelmq.lib.bot import Bot
+from intelmq.lib.exceptions import MissingDependencyError, ConfigurationError
+import time
+
+
+class SplunkSavedSearchBot(Bot):
+
+    is_multithreadable = False
+
+    def init(self):
+        if requests is None:
+            raise MissingDependencyError("requests")
+
+        self.retry_interval = getattr(self.parameters, "retry_interval", 5)
+        self.url = getattr(self.parameters, "url", None)
+        if not self.url:
+            raise ConfigurationError("Connection", "No Splunk API URL specified")
+        self.auth_token = getattr(self.parameters, "auth_token", None)
+        if not self.auth_token:
+            raise ConfigurationError("Connection", "No Splunk API authorization token specified")
+        self.saved_search = getattr(self.parameters, "saved_search", None)
+        if not self.saved_search:
+            raise ConfigurationError("Search", "No Splunk saved search specified")
+
+        self.search_parameters = getattr(self.parameters, "search_parameters", {})
+        self.result_fields = getattr(self.parameters, "result_fields", {})
+
+        self.not_found = getattr(self.parameters, "not_found", ["warn", "send"])
+        if "send" in self.not_found and "drop" in self.not_found:
+            raise ConfigurationError("Processing", "Cannot both drop and send messages without search results")
+
+        self.duplicates = getattr(self.parameters, "not_found", ["warn", "use_first", "send"])
+        if "send" in self.duplicates and "drop" in self.duplicates:
+            raise ConfigurationError("Processing", "Cannot both drop and send messages with duplicate search results")
+        if "ignore" in self.duplicates and "use_first" in self.duplicates:
+            raise ConfigurationError("Processing", "Cannot both ignore and use duplicate search results")
+
+        self.overwrite = getattr(self.parameters, "overwrite", None)
+
+        self.set_request_parameters()
+
+        self.http_header.update({"Authorization": f"Bearer {self.auth_token}"})
```

Comment:
This means that Python 3.5 is not supported by the bot. We need to decide if we drop Python 3.5 in the next IntelMQ release as more and more new contributions use format strings and therefore require Python 3.6

cc @schacht-certat

Reply:
Changed to `"".format()`. Also changed another f-string in a log statement below to %-syntax.